### DISCUSSION TITLE
Backend gzipping fetching

### DISCUSSION THREAD

**Discussion by cyrilchapon** [chimurai/http-proxy-middleware#539] [Q&A]
In here we've accomplished on-the-fly-compression-at-proxy-level with a basic

```
const express = require('express')
const httpProxy = require('http-proxy-middleware')

const app = require('express')()

app.use(compression())
app.use(httpProxy({
  // Our rules here
}))
```

Aggressively optimizing, we now would like to gzip build-time and be able to grab already gzipped files at proxy level, serving them to client

**build**

```
                          bundle.js
myfile.js   =>
                          bundle.js.tar.gz
```

**serve**

```
            /bundle.js/           /bundle.jz.tar.gz
client        =>          proxy          =>           application
```

Anyone with similar experience ?

**Comment by krnlde** [comment]
For me it just works to compress proxy requests. Just put the compression middleware before your proxies and it will determine whether to compress just by their mime types as seen here: https://github.com/expressjs/compression#filter
